DataFormats/Headers/include/Headers/DataHeader.h

```diff
@@ -505,7 +505,12 @@ struct Stack {
   template<typename T, typename... Args>
   static byte* inject(byte* here, const T& h, const Args... args) noexcept {
     auto alsohere = inject(here, h);
-    (reinterpret_cast<BaseHeader*>(here))->flagsNextHeader = true;
+    // the type might be a stack itself, loop through headers and set the flag in the last one
+    byte* nextp = here;
+    while (reinterpret_cast<BaseHeader*>(nextp)->flagsNextHeader) {
+      nextp += reinterpret_cast<BaseHeader*>(nextp)->size();
+    }
+    reinterpret_cast<BaseHeader*>(nextp)->flagsNextHeader = true;
```

Comment:
Hi, at first sight I think a bounds check should not be necessary since the complete buffer size is calculated to be just right...
As for the problem you are trying to solve (thanks for spotting it) I think we need some more type safety there because right now it also takes stuff like arrays and strings: so either restrict the construction of Stack to BaseHeader derived stuff & Stacks AND/OR (in this case) we could add a template specialisation of inject() for Stacks.

Reply:
ah, ok, then we should really specialize (or restrict) to BaseHeader derived stuff and Stacks. Since arrays and strings fit seamlessly into `inject` at the moment, casting those to `BaseHeader` and setting the flag is another bug. I will update the PR.

Reply:
ah, ok, then we should really specialize (or restrict) to BaseHeader derived stuff and Stacks. Since arrays and strings fit seamlessly into inject at the moment, casting those to BaseHeader and setting the flag is another bug. I will update the PR.

Reply:
actually we have to exclude everything else than BaseHeader derived stuff in the Stack. We can not guaranty what is in a Stack when we implement specialized versions of `inject`, and thus we can not set the flags correctly when building a new Stack from additional headers and Stacks. But we rely on that. So, for our purpose, we have to limit Stack, even though it is in principle  more generic.